hicommonwealth/commonwealth
# User-Onboarding: Added preference tags to users profile + community profile page

## Link to Issue
Closes: 
- https://github.com/hicommonwealth/commonwealth/issues/6652
- https://github.com/hicommonwealth/commonwealth/issues/6653

## Description of Changes
- Allows community admins to choose community tags from a select list in the community profile management page `/:community/manage/profile` (API mocked)
- Allows regular users to preference tags from a select list in the edit profile page `/profile/edit` (API mocked)
- Updated the welcome modal preferences section to use the new `<PreferenceTags/>` component 

## "How We Fixed It"
N/A

## Test Plan
- Enable `FLAG_USER_ONBOARDING_ENABLED`
- Login
- Visit `/profile/edit` and scroll to the bottom, verify you see a Preferences section, and you are able to toggle b/w different tags. verify that pressing the save button doesn't cause any unexpected side-effects
- Visit `/:community/manage/profile` (any community profile where you are admin) and scroll to the "Tags" section, verify you are able to 
  - toggle b/w different tags
  - only able to select any 4 tags (max = 4)
  - saving the form doesn't cause any unexpected side-effects
- Turn off the `FLAG_USER_ONBOARDING_ENABLED` flag and verify that tags before this change work correctly.

## Deployment Plan
N/A

## Other Considerations
API for storing selected preferences is not yet ready, so atm that is mocked
